### DISCUSSION TITLE
[Next Js App Directory] Internationalization (i18n) and default language setup:

### DISCUSSION THREAD

**Discussion by AwaisTariq18** [vercel/next.js#46670] [Help]
### Summary

I have implemented localization/Internationalization in next js app directory by following the docs example and setting up the middleware, since i have moved every page under the [lang] directory. its working fine for everything but what I want is whenever the user selects the en language or the default site opens it should show localhost:3000/AnyPageName instead of localhost:3000/en/AnyPageNumber.. is there any way i can do this? need some help here!

**Comment by masterbater** [comment]
Set a cookie in client side and read it. That way you dont need any middleware or subpath
